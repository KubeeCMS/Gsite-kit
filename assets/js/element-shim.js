/**
 * WordPress Element shim.
 *
 * A temporary workaround to ensure the same version of React
 * is always used across multiple entrypoints.
 *
 * Site Kit by Google, Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * @private
 */

import * as element from '@wordpress/element__non-shim';

if ( global.googlesitekit === undefined ) {
	global.googlesitekit = {};
}

const {
	Children,
	cloneElement,
	Component,
	concatChildren,
	createContext,
	createElement,
	createInterpolateElement,
	createPortal,
	createRef,
	findDOMNode,
	forwardRef,
	Fragment,
	isEmptyElement,
	isValidElement,
	lazy,
	memo,
	Platform,
	RawHTML,
	render,
	renderToString,
	StrictMode,
	Suspense,
	switchChildrenNodeName,
	unmountComponentAtNode,
	useCallback,
	useContext,
	useDebugValue,
	useEffect,
	useImperativeHandle,
	useLayoutEffect,
	useMemo,
	useReducer,
	useRef,
	useState,
} = global.googlesitekit._element || element;

export {
	Children,
	cloneElement,
	Component,
	concatChildren,
	createContext,
	createElement,
	createInterpolateElement,
	createPortal,
	createRef,
	findDOMNode,
	forwardRef,
	Fragment,
	isEmptyElement,
	isValidElement,
	lazy,
	memo,
	Platform,
	RawHTML,
	render,
	renderToString,
	StrictMode,
	Suspense,
	switchChildrenNodeName,
	unmountComponentAtNode,
	useCallback,
	useContext,
	useDebugValue,
	useEffect,
	useImperativeHandle,
	useLayoutEffect,
	useMemo,
	useReducer,
	useRef,
	useState,
};

if ( global.googlesitekit._element === undefined ) {
	global.googlesitekit._element = {
		Children,
		cloneElement,
		Component,
		concatChildren,
		createContext,
		createElement,
		createInterpolateElement,
		createPortal,
		createRef,
		findDOMNode,
		forwardRef,
		Fragment,
		isEmptyElement,
		isValidElement,
		lazy,
		memo,
		Platform,
		RawHTML,
		render,
		renderToString,
		StrictMode,
		Suspense,
		switchChildrenNodeName,
		unmountComponentAtNode,
		useCallback,
		useContext,
		useDebugValue,
		useEffect,
		useImperativeHandle,
		useLayoutEffect,
		useMemo,
		useReducer,
		useRef,
		useState,
	};
}
