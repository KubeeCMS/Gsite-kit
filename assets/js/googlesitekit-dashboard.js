/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render, Fragment } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { clearWebStorage, loadTranslations } from './util';
import Root from './components/root';
import './modules';
import Setup from './components/setup/setup-wrapper';
import DashboardApp from './components/dashboard/dashboard-app';
import NotificationCounter from './components/notifications/notification-counter';
import './components/notifications';

const GoogleSitekitDashboard = () => {
	const { showModuleSetupWizard } = global._googlesitekitLegacyData.setup;

	if ( showModuleSetupWizard ) {
		return <Setup />;
	}

	return (
		<Fragment>
			<NotificationCounter />
			<DashboardApp />
		</Fragment>
	);
};

// Initialize the app once the DOM is ready.
domReady( () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById( 'js-googlesitekit-dashboard' );

	if ( renderTarget ) {
		loadTranslations();

		render( <Root dataAPIContext="Dashboard"><GoogleSitekitDashboard /></Root>, renderTarget );
	}
} );
