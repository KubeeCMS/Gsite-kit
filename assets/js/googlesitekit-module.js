/**
 * Module component.
 *
 * Site Kit by Google, Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import './modules';

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { applyFilters } from '@wordpress/hooks';
import { Component, render, Suspense, lazy } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { loadTranslations } from './util';
import ProgressBar from './components/progress-bar';
import './components/data';
import './components/notifications';
import Root from './components/root';
import ModuleApp from './components/module-app';

class GoogleSitekitModule extends Component {
	constructor( props ) {
		super( props );

		this.state = {
			showModuleSetupWizard: global._googlesitekitLegacyData.setup.showModuleSetupWizard,
		};
	}

	render() {
		const {
			showModuleSetupWizard,
		} = this.state;

		const { currentAdminPage } = global._googlesitekitLegacyData.admin;

		/**
		 * Filters whether to show the Module setup wizard when showModuleSetupWizard is true.
		 *
		 * Modules can opt out of the wizard setup flow by returning false.
		 */
		const moduleHasSetupWizard = applyFilters( 'googlesitekit.moduleHasSetupWizard', true, currentAdminPage );

		if ( showModuleSetupWizard && moduleHasSetupWizard ) {
			// Set webpackPublicPath on-the-fly.
			if ( global._googlesitekitLegacyData && global._googlesitekitLegacyData.publicPath ) {
				// eslint-disable-next-line no-undef
				__webpack_public_path__ = global._googlesitekitLegacyData.publicPath; /*eslint camelcase: 0*/
			}

			const Setup = lazy( () => import( /* webpackChunkName: "chunk-googlesitekit-setup-wrapper" */'./components/setup/setup-wrapper' ) );

			return (
				<Suspense fallback={
					<div className="googlesitekit-setup">
						<div className="mdc-layout-grid">
							<div className="mdc-layout-grid__inner">
								<div className="
									mdc-layout-grid__cell
									mdc-layout-grid__cell--span-12
								">
									<div className="googlesitekit-setup__wrapper">
										<div className="mdc-layout-grid">
											<div className="mdc-layout-grid__inner">
												<div className="
													mdc-layout-grid__cell
													mdc-layout-grid__cell--span-12
												">
													<ProgressBar />
												</div>
											</div>
										</div>
									</div>
								</div>
							</div>
						</div>
					</div>
				}>
					<Setup />
				</Suspense>
			);
		}

		return <ModuleApp />;
	}
}

// Initialize the app once the DOM is ready.
domReady( () => {
	const renderTarget = document.getElementById( 'js-googlesitekit-module' );

	if ( renderTarget ) {
		loadTranslations();

		render(
			<Root dataAPIContext="Single" dataAPIModuleArgs={ global.googlesitekitCurrentModule } >
				<GoogleSitekitModule />
			</Root>,
			renderTarget
		);
	}
} );

