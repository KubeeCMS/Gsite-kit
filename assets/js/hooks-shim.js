/**
 * WordPress Hooks API.
 *
 * A temporary workaround to share the same instance of hooks
 * across multiple entrypoints.
 *
 * Site Kit by Google, Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * @private
 */

/**
 * WordPress dependencies
 */
import * as hooks from '@wordpress/hooks__non-shim';

if ( global.googlesitekit === undefined ) {
	global.googlesitekit = {};
}

const {
	addAction,
	addFilter,
	removeAction,
	removeFilter,
	hasAction,
	hasFilter,
	removeAllActions,
	removeAllFilters,
	doAction,
	applyFilters,
	currentAction,
	currentFilter,
	doingAction,
	doingFilter,
	didAction,
	didFilter,
	actions,
	filters,
} = global.googlesitekit._hooks || hooks;
// Ensure exports always reference the same instance.

export {
	addAction,
	addFilter,
	removeAction,
	removeFilter,
	hasAction,
	hasFilter,
	removeAllActions,
	removeAllFilters,
	doAction,
	applyFilters,
	currentAction,
	currentFilter,
	doingAction,
	doingFilter,
	didAction,
	didFilter,
	actions,
	filters,
};

if ( global.googlesitekit._hooks === undefined ) {
	global.googlesitekit._hooks = {
		addAction,
		addFilter,
		removeAction,
		removeFilter,
		hasAction,
		hasFilter,
		removeAllActions,
		removeAllFilters,
		doAction,
		applyFilters,
		currentAction,
		currentFilter,
		doingAction,
		doingFilter,
		didAction,
		didFilter,
		actions,
		filters,
	};
}
