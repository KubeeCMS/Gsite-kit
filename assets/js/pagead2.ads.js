/**
 * This file is intended to detect active ad blocker.
 *
 * Ad blockers block URLs containing the word "ads.js" including this file. The
 * popular AdBlock extension lets that pass though, hence the filename also
 * includes "pagead2" which is blocked by AdBlock.
 *
 * If the file does load, `googlesitekit.canAdsRun` is set to true. If the
 * AdSense datastore is loaded, an action to flag the adblocker inactive will
 * be dispatched.
 */

if ( global._googlesitekitLegacyData === undefined ) {
	global._googlesitekitLegacyData = {};
}

global._googlesitekitLegacyData.canAdsRun = true;

// Ensure that this flag does not get wiped at a later stage during pageload.
document.addEventListener( 'DOMContentLoaded', function() {
	global._googlesitekitLegacyData.canAdsRun = true;
} );

// If registry and AdSense datastore are loaded, use that instead of the global.
// eslint-disable-next-line no-unused-expressions
global.googlesitekit?.data?.dispatch?.( 'modules/adsense' )?.receiveIsAdBlockerActive( false );
