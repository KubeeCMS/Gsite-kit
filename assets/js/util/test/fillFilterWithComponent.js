/**
 * WordPress dependencies
 */
import { applyFilters, addFilter } from '@wordpress/hooks';

/**
 * Internal dependencies
 */
import { fillFilterWithComponent } from '../';

/**
 * External dependencies
 */
import { render } from '../../../../tests/js/test-utils';

describe( 'fillFilterWithComponent', () => {
	it( 'fills filtered component', () => {
		const filterTester = () => '::added::';

		addFilter( 'googlesitekit.Test', 'googlesitekit.AdSenseModuleSettingsDetails', fillFilterWithComponent( filterTester, {} ) );

		const { container } = render( applyFilters( 'googlesitekit.Test', 'test' )() );
		expect( container.firstChild ).toMatchSnapshot();
	} );
} );
