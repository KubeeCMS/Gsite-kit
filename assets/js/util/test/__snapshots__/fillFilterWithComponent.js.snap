// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fillFilterWithComponent fills filtered component 1`] = `::added::`;
